mava/systems/q_learning/rec_iql.py

```diff
@@ -349,22 +370,38 @@ def update_q(
         # Get the data associated with next_obs
         data_next = jax.tree_map(lambda x: x[:, 1:, ...], data)
 
-        first_reward = data_first.reward
-        next_done = data_next.done
+        obs = data_first.obs
+        term_or_trunc = data_first.term_or_trunc
+        reward = data_first.reward
+        action = data_first.action
+
+        # Variables:
+        # - data_first.next_obs
+        # - data_next.term_or_trunc
+        # - data_next.terminal
+        # all come from the same time step. They are stored and accessed in this way
+        # because of the difference between next_obs and the obs of the next timestep,
+        # a difference which only happens at the end of an episode.
+        next_obs = data_first.next_obs
+        next_term_or_trunc = data_next.term_or_trunc
+        next_terminal = data_next.terminal
```

Comment:
think this is slightly clearer:
```suggestion
        # The three following variables all come from the same time step. 
        # They are stored and accessed in this way because of the `AutoResetWrapper`.
        # At the end of an episode `data_first.next_obs` and `data_next.obs` will be
        # different, which is why we need to store both. Thus `data_first.next_obs`
        # aligns with the `terminal` from `data_next`.
        next_obs = data_first.next_obs
        next_term_or_trunc = data_next.term_or_trunc
        next_terminal = data_next.terminal
```